op-node/rollup/derive/channel_in_reader.go

```diff
@@ -86,11 +87,19 @@ func (cr *ChannelInReader) NextBatch(ctx context.Context) (Batch, error) {
 		cr.NextChannel()
 		return nil, NotEnoughData
 	}
-	if batchData.BatchType == SingularBatchType {
-		return &batchData.SingularBatch, nil
+	if batchData.GetBatchType() == SingularBatchType {
```

Comment:
@ImTei What do you think about using switch case statement here? Other conditionals using batchType uses switch case. Search this string `(*SingularBatch)` in the codebase.